### PR TITLE
Login breakage with oauth users

### PR DESCRIPTION
**Pull request by zachmullen**

### CODE REVIEW

**Comment by cpatrick** [comment]
This LGTM. Other than a small nit I decided not to pick around mixing string styles. Well, I almost decided not to pick it... :smiling_imp:

**Comment by zachmullen** [comment]
Should have made this WIP, I'm adding another little feature on here and fixing one security issue.

**Comment by brianhelba** [comment]
LGTM.

Please also close https://github.com/girder/girder/issues/675 and https://github.com/girder/girder/issues/820 when you merge this.

**Comment by zachmullen** [comment]
That last commit should finish this off, it captures everything mentioned in #983. PTAL

**Comment by zachmullen** [comment]
I reverted the security fix and will do that on a separate topic; it's actually going to be a sizable change and will fix several bugs once it's done.

**Comment by zachmullen** [comment]
@cpatrick You had looked at this earlier, could you take another pass? I removed my later change for the most part.

File: plugins/oauth/plugin_tests/oauth_test.py [hunk 1]
```diff
@@ -247,6 +248,14 @@ def googleMock(url, request):
         self.assertEqual(newUser['firstName'], 'John')
         self.assertEqual(newUser['lastName'], 'Doe')
 
+        # Logging in as Oauth-only user should give reasonable error
+        resp = self.request('/user/authentication',
+                            basicAuth='anotheruser:mypassword')
+        self.assertStatus(resp, 400)
+        self.assertEqual(resp.json['message'], "You don't have a password. "
```
Comment (cpatrick) [inline plugins/oauth/plugin_tests/oauth_test.py]: nit: I'm not sure which irks me more, mixed string styles or `\'`...

**Comment by cpatrick** [comment]
@zachmullen A couple of nits, but otherwise LGTM.

**Comment by zachmullen** [comment]
Thanks for keeping me honest, I fixed the quotes. I'll merge once travis is green

**Pull request merged**